packages/network-controller/tests/NetworkController.test.ts

```diff
@@ -5853,24 +5655,25 @@ function buildProviderConfig(config: Partial<ProviderConfig> = {}) {
 }
 
 /**
- * Builds an object that `createInfuraProvider` returns.
- *
- * @returns The object.
+ * Implements just enough of the block tracker interface to pass the tests but
+ * nothing more.
  */
-function buildFakeInfuraProvider() {
-  return {};
+class FakeBlockTracker extends PollingBlockTracker {
+  public eventNames() {
+    return [];
+  }
 }
 
 /**
- * Builds an object that `Subprovider` returns.
+ * Builds an object that `createInfuraProvider` or `createJsonRpcClient` returns.
  *
+ * @param provider - provider to use if you dont want the defaults
  * @returns The object.
  */
-function buildFakeInfuraSubprovider() {
+function buildFakeClient(provider?: SafeEventEmitterProvider) {
   return {
-    handleRequest(_payload: any, _next: any, _end: any) {
-      // do nothing
-    },
+    provider: provider || buildFakeProvider(),
+    blockTracker: new FakeBlockTracker() as PollingBlockTracker,
```

Comment:
Is it possible to make FakeBlockTracker conform to the PollingBlockTracker interface instead of typecasting this?